File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-10.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0, r= 0;
        int res = 0;
        int maxlen = 0;
        unordered_map<char, int> mp;
        while(r < s.size()){
            mp[s[r]]++;
            maxlen = max(maxlen, mp[s[r]]);
            if(r-l+1-maxlen > k){
                mp[s[l]]--;
                l++;
            }
            res = max(res, r-l+1);
            r++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        mp = Counter(nums)
        res = 0
        count = 1
        for k in sorted(mp):
            if k+1 in mp:
                count+=1
                continue
            res=max(count, res)
            count = 1
        return res